
WD=/media/Data/s.siira/tRNA_Phe_RNAseq/pipeline/2_alignment
GENOME=$WD/mm10.genome

for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	INPUT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	MITOPAIR=`samtools view -@ 12 -b -f 2 -F 256 $INPUT | samtools flagstat -@ 12 - | head -n 1 | cut -d ' ' -f1`
	MITOCOUNT=`expr $MITOPAIR / 2`
	SCALE=`awk -v count=$MITOCOUNT 'BEGIN {print 1000000/count}'`
	OUTPUT=$WD/coverage/${SAMPLE}.RNASeq.rpm.bedgraph

	samtools view -@ 12 -b -f 2 -F 256 $INPUT | bedtools genomecov -bg -scale $SCALE -trackline -trackopts 'name="'${SAMPLE}'"' -ibam - -g $GENOME > $OUTPUT
	gzip $OUTPUT
done



### split
WD=/media/Data/s.siira/tRNA_Phe_RNAseq/pipeline/2_alignment
GENOME=$WD/mm10.genome
IRDS=/media/IRDS/Server/s.siira/tRNA_Phe_RNAseq/pipeline/2_alignment

for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	INPUT=$IRDS/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	INPUT2=$IRDS/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam.bai
	OUTPUT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	OUTPUT2=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam.bai

	mkdir $WD/${SAMPLE}
	cp $INPUT $OUTPUT
	cp $INPUT2 $OUTPUT2
done

for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	INPUT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	MITOPAIR=`samtools view -@ 12 -b -f 2 -F 256 $INPUT | samtools flagstat -@ 12 - | head -n 1 | cut -d ' ' -f1`
	MITOCOUNT=`expr $MITOPAIR / 2`
	SCALE=`awk -v count=$MITOCOUNT 'BEGIN {print 1000000/count}'`
	OUTPUT=$WD/coverage/${SAMPLE}.RNASeq.rpm.split.bedgraph

	samtools view -@ 12 -b -f 2 -F 256 $INPUT | bedtools genomecov -bg -split -scale $SCALE -trackline -trackopts 'name="'${SAMPLE}'"' -ibam - -g $GENOME > $OUTPUT
	gzip $OUTPUT
done


### split, per-strand
## Strand separation
WD=/media/Data/s.siira/tRNA_Phe_RNAseq/pipeline/2_alignment
for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	OUTDIR=$WD/${SAMPLE}
	INPUT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam

	FWD1=$OUTDIR/${SAMPLE}.fwd1.bam
	FWD2=$OUTDIR/${SAMPLE}.fwd2.bam
	FWD=$OUTDIR/${SAMPLE}.fwd.bam

	samtools view -b -f 128 -F 16 $INPUT > $FWD1
	samtools index $FWD1

	samtools view -b -f 80 $INPUT > $FWD2
	samtools index $FWD2

	samtools merge -f $FWD $FWD1 $FWD2
	samtools index $FWD

	rm $FWD1 $FWD1.bai $FWD2 $FWD2.bai

	REV1=$OUTDIR/${SAMPLE}.rev1.bam
	REV2=$OUTDIR/${SAMPLE}.rev2.bam
	REV=$OUTDIR/${SAMPLE}.rev.bam

	samtools view -b -f 144 $INPUT > $REV1
	samtools index $REV1

	samtools view -b -f 64 -F 16 $INPUT > $REV2
	samtools index $REV2

	samtools merge -f $REV $REV1 $REV2
	samtools index $REV

	rm $REV1 $REV1.bai $REV2 $REV2.bai
done

# fwd
for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	INPUT=$WD/${SAMPLE}/${SAMPLE}.fwd.bam
	COUNT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	MITOPAIR=`samtools view -@ 12 -b -f 2 -F 256 $COUNT | samtools flagstat -@ 12 - | head -n 1 | cut -d ' ' -f1`
	MITOCOUNT=`expr $MITOPAIR / 2`
	SCALE=`awk -v count=$MITOCOUNT 'BEGIN {print 1000000/count}'`
	OUTPUT=$WD/coverage/${SAMPLE}.RNASeq.rpm.split.fwd.bedgraph

	samtools view -@ 12 -b -f 2 -F 256 $INPUT | bedtools genomecov -bg -split -scale $SCALE -trackline -trackopts 'name="'${SAMPLE}'"' -ibam - -g $GENOME > $OUTPUT
	gzip $OUTPUT
done

# rev
for SAMPLE in 1_H2G73DSXY_CCGCGGTT-AGCGCTAG 2_H2G73DSXY_TTATAACC-GATATCGA 3_H2G73DSXY_GGACTTGG-CGCAGACG 4_H2G73DSXY_AAGTCCAA-TATGAGTA 5_H2G73DSXY_ATCCACTG-AGGTGCGT 6_H2G73DSXY_GCTTGTCA-GAACATAC 7_H2G73DSXY_CAAGCTAG-ACATAGCG 8_H2G73DSXY_TGGATCGA-GTGCGATA 9_H2G73DSXY_AGTTCAGG-CCAACAGA 10_H2G73DSXY_GACCTGAA-TTGGTGAG 11_H2G73DSXY_TCTCTACT-CGCGGTTC 12_H2G73DSXY_CTCTCGTC-TATAACCT 13_H2G73DSXY_CCAAGTCT-AAGGATGA 14_H2G73DSXY_TTGGACTC-GGAAGCAG 15_H2G73DSXY_GGCTTAAG-TCGTGACC 16_H2G73DSXY_AATCCGGA-CTACAGTT 17_H2G73DSXY_TAATACAG-ATATTCAC 18_H2G73DSXY_CGGCGTGA-GCGCCTGT 19_H2G73DSXY_ATGTAAGT-ACTCTATG 20_H2G73DSXY_GCACGGAC-GTCTCGCA 21_H2G73DSXY_GGTACCTT-AAGACGTC 22_H2G73DSXY_AACGTTCC-GGAGTACT 23_H2G73DSXY_GCAGAATT-ACCGGCCA 24_H2G73DSXY_ATGAGGCC-GTTAATTG
do
	INPUT=$WD/${SAMPLE}/${SAMPLE}.rev.bam
	COUNT=$WD/${SAMPLE}/${SAMPLE}.Aligned.sortedByCoord.out.bam
	MITOPAIR=`samtools view -@ 12 -b -f 2 -F 256 $COUNT | samtools flagstat -@ 12 - | head -n 1 | cut -d ' ' -f1`
	MITOCOUNT=`expr $MITOPAIR / 2`
	SCALE=`awk -v count=$MITOCOUNT 'BEGIN {print 1000000/count}'`
	OUTPUT=$WD/coverage/${SAMPLE}.RNASeq.rpm.split.rev.bedgraph

	samtools view -@ 12 -b -f 2 -F 256 $INPUT | bedtools genomecov -bg -split -scale $SCALE -trackline -trackopts 'name="'${SAMPLE}'"' -ibam - -g $GENOME > $OUTPUT
	gzip $OUTPUT
done



